analysis_scripts/Run_Sim_Plot_Population.R

```diff
@@ -11,6 +11,38 @@ library(tidyverse)
 #############################################################
 rm(list = ls())
 
+get_mortality <- function(results, categoryName, agebands){
+  # Get mortality matrix to data frame
+  # categoryName can be maleMortality, femaleMortality etc
+  # ageband is a dataframe with Label column that matched results' mortality labels and AgeBand column that is custom defined
+  
+  # Access the rateMatrix
+  all_trials_matrix <- 
+    lapply(
+      lapply(
+        lapply(
+          lapply(results, "[[", "PopulationRates"),
+          "[[", categoryName),
+        "[[", "ratesMatrix"),
+      data.frame)
+  
+  # Pivot the rateMatrix year columns to a single column named "Year"
+  all_trials_matrix <- lapply(all_trials_matrix, rownames_to_column, "Label")
+  all_trials_matrix <- lapply(all_trials_matrix, 
+                              function(x){
+                                pivot_longer(x, colnames(x)[grepl("X", colnames(x))], 
+                                             names_prefix = "X", 
+                                             names_to = "Year", 
+                                             values_to = "MortalityRate")
+                              }
+  )
+  # Combine all trials and join to the custom AgeBand
+  all_trials_matrix <- do.call(data.table::rbindlist, list(all_trials_matrix, idcol="Trial"))
+  mortalityrates <- all_trials_matrix %>% 
+    inner_join(agebands)
```

Comment:
Here's a faster way to generate the all_trials_matrix dataframe (albeit with slightly different column and sort orders):

```
trials <- names(results)
l <- lapply(trials, function(trial){
  t <- results[[trial]][["PopulationRates"]][[categoryName]][["ratesMatrix"]]
  t <- reshape2::melt(t, varnames = c("Label", "Year"), value.name = "MortalityRate")
  t$Trial = rep(trial, nrow(t))
  return(t)
})

all_trials_matrix <- data.table::rbindlist(l)
all_trials_matrix$Label <- as.character(all_trials_matrix$Label)
all_trials_matrix$Year <- as.character(all_trials_matrix$Year)
```